src/gt4py/next/iterator/transforms/collapse_tuple.py

```diff
@@ -25,8 +31,72 @@ def _get_tuple_size(type_: type_inference.Type) -> int:
     )
     return len(type_.dtype)
 
+def _is_let(node: ir.Node) -> bool:
+    return isinstance(node, ir.FunCall) and isinstance(node.fun, ir.Lambda)
 
-@dataclass(frozen=True)
+def _is_if_call(node: ir.Expr):
+    return isinstance(node, ir.FunCall) and node.fun == im.ref("if_")
+
+def _with_altered_arg(node: ir.FunCall, arg_idx: int, new_arg: ir.Expr):
+    return ir.FunCall(
+        fun=node.fun,
+        args=[arg if i != arg_idx else new_arg for i, arg in enumerate(node.args)]
+    )
+
+def _is_trivial_make_tuple_call(node: ir.Expr):
+    if not (isinstance(node, ir.FunCall) and node.fun == im.ref("make_tuple")):
+        return False
+    if not all(isinstance(arg, (ir.SymRef, ir.Literal)) or _is_trivial_make_tuple_call(arg) for arg in node.args):
+        return False
+    return True
+
+def nlet(bindings: list[tuple[ir.Sym, str], ir.Expr]):
+    return im.let(*[el for tup in bindings for el in tup])
+
+def _short_hash(val: str) -> str:
+    return hashlib.sha1(val.encode('UTF-8')).hexdigest()[0:6]
+
+@dataclasses.dataclass(frozen=True)
+class CannonicalizeBoundSymbolNames(eve.NodeTranslator):
```

Comment:
```suggestion
class CanonicalizeBoundSymbolNames(eve.NodeTranslator):
```

Reply:
I know it's in draft, but I'll still leave comments where I think stuff should go: Should go in a separate file, because it will be hard to find this here if we have a similar use-case later.